ruby/ruby/IceLocal/LocalExceptions.rb

```diff
@@ -0,0 +1,159 @@
+# encoding: utf-8
+#
+# Copyright (c) ZeroC, Inc.
+
+# This file provides all the exceptions derived from Ice::LocalException.
+
+module Ice
+    #
+    # The 6 (7 with the RequestFailedException base class) special local exceptions that can be marshaled in an Ice
+    # reply message. Other local exceptions can't be marshaled.
+    #
+
+    class RequestFailedException < LocalException
+        def initialize(id, facet, operation, msg)
+            super(msg)
+            @id = id
+            @facet = facet
+            @operation = operation
+        end
+
+        attr_reader :id, :facet, :operation
+    end
+
+    class ObjectNotExistException < RequestFailedException
+    end
+
+    class FacetNotExistException < RequestFailedException
+    end
+
+    class OperationNotExistException < RequestFailedException
+    end
+
+    class UnknownException < LocalException
+    end
+
+    class UnknownLocalException < UnknownException
+    end
+
+    class UnknownUserException < UnknownException
+    end
+
+    #
+    # Protocol exceptions
+    #
+
+    class ProtocolException < LocalException
+    end
+
+    class CloseConnectionException < ProtocolException
+    end
+
+    class ConnectionManuallyClosedException < LocalException
+    end
+
+    class DatagramLimitException < ProtocolException
+    end
+
+    class MarshalException < ProtocolException
+    end
+
+    #
+    # Timeout exceptions
+    #
+
+    class TimeoutException < LocalException
+    end
+
+    class ConnectTimeoutException < TimeoutException
+    end
+
+    class CloseTimeoutException < TimeoutException
+    end
+
+    class InvocationTimeoutException < TimeoutException
+    end
+
+    #
+    # Syscall exceptions
+    #
+
+    class SyscallException < LocalException
+    end
+
+    class DNSException < SyscallException
+    end
+
+    #
+    # Socket exceptions
+    #
+
+    class SocketException < SyscallException
+    end
+
+    class ConnectFailedException < SocketException
+    end
+
+    class ConnectionLostException < SocketException
+    end
+
+    class ConnectionRefusedException < ConnectFailedException
+    end
+
+    #
+    # Other leaf local exceptions in alphabetical order.
+    #
+
+    class AlreadyRegisteredException < LocalException
+        def initialize(kindOfObject, id, msg)
+            super(msg)
+            @kindOfObject = kindOfObject
+            @id = id
+        end
+
+        attr_reader :kindOfObject, :id
+    end
+
+    class CommunicatorDestroyedException < LocalException
+    end
+
+    class ConnectionIdleException < LocalException
+    end
+
+    class FeatureNotSupportedException < LocalException
+    end
+
+    class FixedProxyException < LocalException
+    end
+
+    class InitializationException < LocalException
+    end
+
+    class InvocationCanceledException < LocalException
+    end
+
+    class NoEndpointException < LocalException
+    end
+
+    class NotRegisteredException < LocalException
```

Comment:
This exception is thrown by the client runtime when a locator resolution fails - and we have a Ruby test for it. Need to add this exception back in MATLAB.